spec/GraphQLOverHTTP.md

```diff
@@ -745,3 +745,37 @@ payload is `application/json` then the client MUST NOT rely on the body to be a
 well-formed _GraphQL response_ since the source of the response may not be the
 server but instead some intermediary such as API gateways, proxies, firewalls,
 etc.
+
+# Additional Notes
+
+## Security
+
+In this specification, GET requests are not supported for mutations due to
+security concerns. GET requests expose variables to logging mechanisms and
+intermediaries due to the URL encoding of parameters, which can lead to
+sensitive data being inadvertently logged. Furthermore, GET requests are
+considered "simple requests" under CORS (Cross-Origin Resource Sharing), meaning
+they bypass preflight checks that add a layer of security.
+
+On the other hand, using `application/json` for request bodies mandates a CORS
+preflight request, adding a security layer by ensuring the client has explicit
+permission from the server before sending the actual request. This is
+particularly important in mitigating cross-site request forgery (CSRF) attacks.
+
+Additionally, supporting form data requests (`application/x-www-form-urlencoded`
+or `multipart/form-data`) could pose significant security risks. Form data
+requests may be vulnerable to CSRF and other attacks due to the lack of CORS
+preflight checks. As a result, the use of form data for GraphQL queries or
+mutations is discouraged.
+
+For more detailed security considerations, please refer to
+[RFC 7231](https://tools.ietf.org/html/rfc7231),
+[RFC 6454](https://tools.ietf.org/html/rfc6454), and other relevant RFCs.
+
+## Format Compatibility
+
+Supporting formats not described by this specification, such as XML or Protobuf,
+may conflict with future versions of this specification, as ongoing development
+aims to standardize and ensure the security and interoperability of GraphQL over
+HTTP. For this reason, it is recommended to adhere to the officially recognized
+formats outlined here.
```

Comment:
+1 on the concern but would also like to encourage experimentations with other encodings since JSON is quite ineffective (see also https://github.com/graphql/graphql-spec/issues/432).

We also have language in the current spec that alternative formats are supported ([source](https://github.com/graphql/graphql-over-http/blob/68d029773271278c7cf5ea2545ab303829b0db59/spec/GraphQLOverHTTP.md?plain=1#L167)):

```
Servers and clients MUST support JSON and MAY support other, additional
serialization formats.
```

So the current text somewhat contradicts that.

What about we reserve all `graphql-response+*` to the GraphQL spec and users can use the rest of the namespace for experimentation?

Reply:
I the concern relates to request formats moreso than response formats.  Presumably if the client uses a nonstandard request format, they will already have an expectation of the response format.  If there becomes an additional standard to the spec that creates a conflict in the request format, the server may not understand the request or respond in the format the client expects.  So I think the conflict with future request formats is what needs to be addressed here.

Reply:
Gotcha 👍 Then maybe make it explicit that this is about requests? 

```md
## Request body media type compatibility

For request bodies, supporting media types not described in this specification, such as XML, ...
```

Reply:
I've taken another pass at this paragraph.  Please review the revised PR - thanks!